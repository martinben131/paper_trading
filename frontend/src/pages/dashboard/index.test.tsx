import React from "react";
import { shallow } from "enzyme";
import Dashboard from "./index";

jest.mock("../../store", () => ({
  useProfile: () => [
    {
      ownedStocks: [],
      profit: 50,
      balance: 100,
    },
  ],
  useStocks: () => [{}],
}));

jest.mock("react-router-dom", () => ({
  useHistory: () => ({}),
}));

test("Renders dashboard correctly", () => {
  const component = shallow(<Dashboard />);
  expect(component).toMatchSnapshot();
});
