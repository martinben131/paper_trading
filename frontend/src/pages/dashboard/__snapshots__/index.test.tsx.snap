// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders dashboard correctly 1`] = `
<Blueprint3.NonIdealState
  action={
    <Blueprint3.Button
      intent="success"
      onClick={[Function]}
      rightIcon="arrow-right"
      text="See all stocks"
    />
  }
  description="Visit the stocks page to buy some stocks. You have $100.00 to buy stocks with."
  icon="clean"
  title="You don't own any stocks"
/>
`;
